pkg/agent/catalog/catalog.go

```diff
@@ -37,24 +37,24 @@ type Catalog interface {
 
 var (
 	supportedPlugins = map[string]goplugin.Plugin{
-		KeyManagerType:       &keymanager.KeyManagerPlugin{},
-		NodeAttestorType:     &nodeattestor.NodeAttestorPlugin{},
-		WorkloadAttestorType: &workloadattestor.WorkloadAttestorPlugin{},
+		KeyManagerType:       &keymanager.KeyManagerGRPCPlugin{},
+		NodeAttestorType:     &nodeattestor.NodeAttestorGRPCPlugin{},
+		WorkloadAttestorType: &workloadattestor.WorkloadAttestorGRPCPlugin{},
 	}
 
 	builtinPlugins = common.BuiltinPluginMap{
 		KeyManagerType: {
-			"disk":   disk.New(),
-			"memory": memory.New(),
+			"disk":   keymanager.NewKeyManagerBuiltIn(disk.New()),
+			"memory": keymanager.NewKeyManagerBuiltIn(memory.New()),
```

Comment:
Maybe `NewKeyManagerBuiltIn` -> `NewBuiltIn`? The keymanager bit is already specified via package name

Reply:
yeah, i considered that but wasn't sure if we'd ever have two services under the same package. If we're committed to only a single service, then the name can be shortened for sure.